src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -516,6 +527,10 @@ public Task<IActionResult> EditPOST(string contentItemId, string returnUrl)
             {
                 return RedirectToAction("Edit", new RouteValueDictionary { { "ContentItemId", contentItem.ContentItemId } });
             }
+            else if (stayOnSamePage)
```

Comment:
I don't think this code is necessary if `returnUrl` is computed from the caller. Because even with _continue_ we redirect. We just need to set the current page to returnUrl and maybe add the previous returnUrl to the query string.